src/nps/userSurvey.ts

```diff
@@ -38,83 +39,24 @@ export class UserSurvey {
-
-        // If the user has opted out of the survey, don't prompt for feedback
-        const isNeverUser = globalState.get(NEVER_KEY, false);
-        if (isNeverUser) {
-            return;
-        }
-
-        // If the user has already been prompted for feedback in this version, don't prompt again
-        const extensionVersion =
-            vscode.extensions.getExtension(constants.extensionId).packageJSON
-                .version || "unknown";
-        const skipVersion = globalState.get(SKIP_VERSION_KEY, "");
-        if (skipVersion === extensionVersion) {
-            return;
-        }
-
-        const date = new Date().toDateString();
-        const lastSessionDate = globalState.get(
-            LAST_SESSION_DATE_KEY,
-            new Date(0).toDateString(),
-        );
-
-        if (date === lastSessionDate) {
-            return;
-        }
-
         const sessionCount = globalState.get(SESSION_COUNT_KEY, 0) + 1;
-        await globalState.update(LAST_SESSION_DATE_KEY, date);
-        await globalState.update(SESSION_COUNT_KEY, sessionCount);
-
-        // don't prompt for feedback from users until they've had a chance to use the extension a few times
-        if (sessionCount < 5) {
-            return;
-        }
-
-        const isCandidate =
-            globalState.get(IS_CANDIDATE_KEY, false) ||
-            Math.random() < PROBABILITY;
-
-        await globalState.update(IS_CANDIDATE_KEY, isCandidate);
 
-        if (!isCandidate) {
-            await globalState.update(SKIP_VERSION_KEY, extensionVersion);
+        if (true && !this.shouldPromptForFeedback()) {
             return;
         }
 
         const take = {
             title: locConstants.UserSurvey.takeSurvey,
             run: async () => {
                 const state: UserSurveyState = getStandardNPSQuestions();
-                if (
-                    !this._webviewController ||
-                    this._webviewController.isDisposed
-                ) {
-                    this._webviewController = new UserSurveyWebviewController(
-                        this._context,
-                        state,
-                    );
-                } else {
-                    this._webviewController.updateState(state);
-                }
-                this._webviewController.revealToForeground();
-
-                const answers = await new Promise<Answers>((resolve) => {
-                    this._webviewController.onSubmit((e) => {
-                        resolve(e);
-                    });
-
-                    this._webviewController.onCancel(() => {
-                        resolve({});
-                    });
-                });
-                sendSurveyTelemetry("nps", answers);
-                await globalState.update(IS_CANDIDATE_KEY, false);
-                await globalState.update(SKIP_VERSION_KEY, extensionVersion);
+                await this.launchSurvey("nps", state);
+
+                // TODO: revert before PR
```

Comment:
Just fyi @Benjin , there is a vscode command "User feedback" that launches the feedback form bypassing these checks

Reply:
I tried the `vscode., and all it does is prepopulate the github form itself (unless I missed an option?).  If I used that, then the user's flow would be:

Reply:
Scratch my comment; you were referring to the MSSQL command you added!